XRTK-Core/Packages/com.xrtk.core/Tests/Services/TestService1.cs

```diff
@@ -24,5 +33,26 @@ public override void Disable()
         {
             IsEnabled = false;
         }
+
+        public bool RegisterDataProvider(ITestDataProvider1 dataProvider)
```

Comment:
I'm wondering if there's a better way to do this.

I don't really like the pattern we're using even in the real services and data providers. Any data provider already registered under the service should technically already be known about. As long as it matches one of the pre-registered types why does it need to register itself again, ya know?

Reply:
That was my thinking, I hate boilerplate, as this registration pattern (which is valid, data providers should announce themselves to their service), but is there a better way so we don't need to declare it everytime

Reply:
This needs to be addressed in a separate PR